src/atomate2/common/flows/qha.py

```diff
@@ -0,0 +1,189 @@
+        Maker to relax deformed structures for the EOS fit.
+        The volume has to be fixed!
+    phonon_displacement_maker: .ForceFieldStaticMaker | .BaseVaspMaker | None
+    phonon_static_maker: .ForceFieldStaticMaker | .BaseVaspMaker | None
+    phonon_maker_kwargs: dict
+    linear_strain: tuple[float, float]
+        Percentage linear strain to apply as a deformation, default = -5% to 5%.
+    number_of_frames: int
+        Number of strain calculations to do for EOS fit, default = 6.
+    t_max: float | None
+        Maximum temperature until which the QHA will be performed
+    pressure: float | None
+        Pressure at which the QHA will be performed (default None, no pressure)
+    skip_analysis: bool
+        Skips the analysis step and only performs EOS and phonon computations.
+    ignore_imaginary_modes: bool
+        By default, volumes where the harmonic phonon approximation shows imaginary
+        will be ignored
+    eos_type: str
+        Equation of State type used for the fitting. Defaults to vinet.
+    """
+
+    name: str = "QHA Maker"
+    initial_relax_maker: ForceFieldRelaxMaker | BaseVaspMaker | None = None
+    eos_relax_maker: ForceFieldRelaxMaker | BaseVaspMaker | None = None
+    phonon_displacement_maker: ForceFieldStaticMaker | BaseVaspMaker | None = None
+    phonon_static_maker: ForceFieldStaticMaker | BaseVaspMaker | None = None
+    phonon_maker_kwargs: dict = field(default_factory=dict)
+    linear_strain: tuple[float, float] = (-0.05, 0.05)
+    number_of_frames: int = 6
+    t_max: float | None = None
+    pressure: float | None = None
+    ignore_imaginary_modes: bool = False
+    skip_analysis: bool = False
+    eos_type: Literal["vinet", "birch_murnaghan", "murnaghan"] = "vinet"
+    analyze_free_energy_kwargs: dict = field(default_factory=dict)
+    # TODO: implement advanced handling of
+    #  imaginary modes in phonon runs (i.e., fitting procedures)
+
+    def make(self, structure: Structure, prev_dir: str | Path = None) -> Flow:
+        """Run an EOS flow.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        prev_dir : str or Path or None
+            A previous calculation directory to copy output files from.
+
+        Returns
+        -------
+        .Flow, a QHA flow
+        """
+        if self.eos_type not in supported_eos:
+            raise ValueError(
+                "EOS not supported.",
+                "Please choose 'vinet', 'birch_murnaghan', 'murnaghan'",
+            )
+
+        qha_jobs = []
+
+        # In this way, one can easily exchange makers and enforce postprocessor None
+        self.eos = CommonEosMaker(
+            initial_relax_maker=self.initial_relax_maker,
+            eos_relax_maker=self.eos_relax_maker,
+            static_maker=None,
+            postprocessor=None,
+            number_of_frames=self.number_of_frames,
+        )
```

Comment:
@utf I am currently wondering about how to initalize and reuse the EosMaker in the best way. Unfortunately, starting  solely from the EosMaker for the qha workflow would take too much time.

Reply:
@utf ? any opinions?

Reply:
I'm a little confused - what do you mean by:
> starting solely from the EosMaker for the qha workflow would take too much time

Either way, I'm open to both approaches:
- Adding the EosMaker as an argument would reduce the number of options in the `CommonQhaMaker` class but make it more difficult to set properties such as the number of frames. I don't think this is too much of an issue if we could include an example of how to configure the settings in the docstring/documentation page.
- The current internal use of CommonEosMaker makes it easier to configure settings but is slightly less clean code. By the way, I noticed the `linear_strain` option is not currently used.

I'll let you decide!